src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -1246,12 +1246,53 @@ module Crystal
       end
 
       # Check if it's ->LibFoo.foo, so we deduce the type from that method
-      if node.args.empty? && obj && (obj_type = obj.type).is_a?(LibType)
-        matching_fun = obj_type.lookup_first_def(node.name, false)
-        node.raise "undefined fun '#{node.name}' for #{obj_type}" unless matching_fun
+      if obj.type.is_a?(LibType)
+        matching_fun = obj.type.lookup_first_def(node.name, false).as(External?)
```

Comment:
Shouldn't this use `as?` instead?

```suggestion
        matching_fun = obj.type.lookup_first_def(node.name, false).as?(External?)
```

Reply:
https://github.com/crystal-lang/crystal/blob/8850cf7b4588c80143012563cafb7603a2a6ba9f/src/compiler/crystal/semantic/lib.cr#L7